src/configuration/from_file_spec.ts

```diff
@@ -0,0 +1,79 @@
+import { expect } from 'chai'
+import { promisify } from 'util'
+import fs from 'fs'
+import tmp, { DirOptions } from 'tmp'
+import path from 'path'
+import { fromFile } from './from_file'
+
+async function setup(
+  file: string = 'cucumber.js',
+  content: string = `module.exports = {default: {paths: ['some/path/*.feature']}, p1: {paths: ['other/path/*.feature']}, p2: 'other/other/path/*.feature --no-strict'}`
+) {
+  const cwd = await promisify<DirOptions, string>(tmp.dir)({
+    unsafeCleanup: true,
+  })
+  fs.writeFileSync(path.join(cwd, file), content, { encoding: 'utf-8' })
+  return { cwd }
+}
+
+describe('fromFile', () => {
+  it('should return empty config if no default provide and no profiles requested', async () => {
+    const { cwd } = await setup(
+      'cucumber.js',
+      `module.exports = {p1: {paths: ['other/path/*.feature']}}`
+    )
+
+    const result = await fromFile(cwd, 'cucumber.js', [])
+    expect(result).to.deep.eq({})
+  })
+
+  it('should get default config from file if no profiles requested', async () => {
+    const { cwd } = await setup()
+
+    const result = await fromFile(cwd, 'cucumber.js', [])
+    expect(result).to.deep.eq({ paths: ['some/path/*.feature'] })
+  })
+
+  it('should throw when a requested profile doesnt exist', async () => {
+    const { cwd } = await setup()
+
+    try {
+      await fromFile(cwd, 'cucumber.js', ['nope'])
+      expect.fail('should have thrown')
+    } catch (error) {
+      expect(error.message).to.eq(`Requested profile "nope" doesn't exist`)
+    }
+  })
+
+  it('should get single profile config from file', async () => {
```

Comment:
Would it be possible to add some checks with other types of file, like json, or esm?
Or do we rely on acceptance tests for that?

Reply:
Fair, I'll add some more tests here.